lib/Sema/TypeCheckAttr.cpp

```diff
@@ -7216,7 +7216,19 @@ void AttributeChecker::visitNonisolatedAttr(NonisolatedAttr *attr) {
     }
   }
 
+  // Handler values other than variable declarations and constructors.
   if (auto VD = dyn_cast<ValueDecl>(D)) {
+    if (attr->isUnsafe()) {
```

Comment:
You also need to check that `VD` is not a `VarDecl` that has storage (via `AbstractStorageDecl::hasStorage()` which `VarDecl` inherits)

Reply:
Yeah just commented up there that it's wrong actually but had to leave the house 😅

Will fix this soon :)